src/functions/dispatchOpportunityProcessedResult/handler.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  DeleteObjectsCommand,
+  GetObjectCommand,
+  GetObjectCommandInput,
+  GetObjectCommandOutput,
+  ListObjectsCommand,
+  ListObjectsCommandInput,
+  S3Client,
+} from "@aws-sdk/client-s3";
+import { fromTemporaryCredentials } from "@aws-sdk/credential-providers";
+import { opportunityCreatedEvent } from "@libs/event";
+import { OpportunityResult } from "@libs/types";
+import { Readable } from "stream";
+
+const s3Client = new S3Client({
+  region: "us-west-2",
+  credentials: fromTemporaryCredentials({
+    params: {
+      RoleArn: process.env.ACE_ASSUME_ROLE_ARN,
+      RoleSessionName: "ACE_session",
+    },
+  }),
+});
+
+export const main = async (): Promise<void> => {
+  // Get all file names since last time
+  const processedOpportunities = await listProcessedInboundOpportunities();
+  console.log("Processed Opportunities Files", processedOpportunities);
+
+  if (processedOpportunities.length === 0) {
+    console.log("Nothing new to process");
+
+    return;
+  }
+
+  const publishedEvents = await publishEvents(processedOpportunities);
+  console.log(`Successfully published ${publishedEvents.length}`);
+  const deleteObjectsResponse = await deleteReadObjects(processedOpportunities);
+  console.log(
+    "Successfully deleted read objects from bucket",
+    deleteObjectsResponse
+  );
+
+  return;
+};
+
+const listProcessedInboundOpportunities = async (): Promise<string[]> => {
+  const prefix = "opportunity-inbound-processed-results/";
+  const listInput: ListObjectsCommandInput = {
+    Bucket: process.env.BUCKET_NAME,
+    Prefix: prefix,
+  };
+  const listCommand = new ListObjectsCommand(listInput);
+
+  const processedOpportunities = await s3Client.send(listCommand);
+  console.log(processedOpportunities);
+
+  return processedOpportunities.Contents.map((file) => file.Key).filter(
+    (key) => key !== `${prefix}/`
+  );
+};
+
+const publishEvents = async (fileKeys: string[]) => {
+  const events = await Promise.all(
```

Comment:
tu peux modifier un peu cette strat : sur tous les calls que tu fais, y a une chance non négligeable pour que certains fails.
- emet l'évènement sans lire le contenu, juste avec la clé du fichier
- fais un seul call à EventBridge plutôt qu'autant qu'il y a de message https://github.com/fredericbarthelet/typebridge#use-the-event-class-to-create-an-event
- la méthode putEvents de EventBridge ne fail pas si certains messages n'ont pas réussi à être publié dans le bus, c'est la valeur retournée par cette méthode qui te donne des infos sur la quantité de message en erreur à potentiellement réessayer
- déporte la suppression du fichier dans la lambda qui fait le call à hubspot pour mettre à jour l'id d'APN